src/FtpExtractor.php

```diff
@@ -56,14 +55,8 @@ public function copyFiles(string $sourcePath, string $destinationPath, FileState
             $adapter = $this->ftpFilesystem->getAdapter();
             $adapter->getConnection();
             $this->logger->info("Connected to host");
-        } catch (\RuntimeException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
-        } catch (\LogicException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
-        } catch (\ErrorException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
-        } catch (FileNotFoundException $e) {
-            throw new UserException($e->getMessage(), $e->getCode(), $e);
+        } catch (\RuntimeException | \LogicException | \ErrorException | FileNotFoundException $e) {
```

Comment:
Jakou má ten convertor hodnotu, když do něj spadnou stejně vždycky jen ty exceptiony, které konvertuje? Klidně by tam mohlo být místo něj `throw new UserException($e->getMessage(), $e->getCode(), $e);` a nic by se nezměnilo. 

Za mě by měl mít několik "handle" metod. Třeba `handleCopyFilesException($e)` nebo `handlePrepareToDownloadFolderException($e)` a v těch by měly být zkontrolované ty konkrétní výjimky. Ale dostat by měl jakoukoli a jen pokud ji nezná, tak ji vyhodí zabalenou dál jako aplikační...

Reply:
A úplně nejvíc mě teď napadlo, že bysme mohli zkusit poslat PR přímo do toho adapteru a vyměnit tam `throw new LogicException` za `throw new ConnectionErrorException()` přičemž `class ConnectionErrorException implements SftpAdapterException extends \LogicException`. Tím pádem by to bylo zpětně kompatibilní a jednak by se daly chytat oběcně chyby z toho adapteru (skrz ten interface) a jednak by člověk mohl rovnou chytat ty konkrétní. 

Mám to udělat nebo chceš ty?

Reply:
Udělám já